ocaml/database/redo_log.ml

```diff
@@ -265,26 +263,30 @@ let connect sockpath latest_response_time =
       Unix.connect s (Unix.ADDR_UNIX sockpath) ;
       D.debug "Connected to I/O process via socket %s" sockpath ;
       s
-    with Unix.Unix_error (a, b, _) ->
+    with Unix.Unix_error (a, b, _) -> (
       (* It's probably the case that the process hasn't started yet. *)
       (* See if we can afford to wait and try again *)
       Unix.close s ;
       let attempt_delay = !Db_globs.redo_log_connect_delay in
-      let now = Unix.gettimeofday () in
-      let remaining = latest_response_time -. now in
-      if attempt_delay < remaining then (
-        (* Wait for a while then try again *)
-        D.debug
-          "Waiting to connect to I/O process via socket %s (error was %s: \
-           %s)..."
-          sockpath b (Unix.error_message a) ;
-        D.debug "Remaining time for retries: %f" remaining ;
-        Thread.delay attempt_delay ;
-        attempt ()
-      ) else (
-        D.debug "%s timeout reached" __FUNCTION__ ;
-        raise Unixext.Timeout
-      )
+      match
+        Clock.Timer.remaining
+          (Clock.Timer.shorten_by attempt_delay latest_response_time)
+      with
+      | Remaining remaining ->
+          (* Wait for a while then try again *)
+          D.debug
+            "Waiting to connect to I/O process via socket %s (error was %s: \
+             %s)..."
+            sockpath b (Unix.error_message a) ;
+          D.debug "Remaining time for retries: %s"
+            (Fmt.to_to_string Mtime.Span.pp remaining) ;
+          Unixext.delay_span attempt_delay ;
+          attempt ()
+      | Expired overrun ->
+          D.debug "%s timeout reached (%ss overrun)" __FUNCTION__
+            (Fmt.to_to_string Mtime.Span.pp overrun) ;
```

Comment:
```suggestion
          D.debug "%s timeout reached (%a overrun)" __FUNCTION__
            Debug.Pp.mtime_span overrun ;
```

Reply:
Applied, this also fixes the extra `s` at the end which Mtime's printer might've already added.